### DISCUSSION TITLE
Temperature stability of the LibreVNA device (stable measurements).

### DISCUSSION THREAD

**Discussion by YuryW** [jankae/LibreVNA#192] [Ideas]
Hi
Idea for Jankae.
Make an external "oven" for LibreVNA according to the already mastered principle of maintaining a constant temperature in LibreCal. This is for those users who have already purchased LibreVNA and need accurate temperature independent measurements. The appearance of the stove is an aluminum plate, fastened with four screws from the bottom of the LibreVNA into the existing threads (four regular screws are unscrewed before that). Its purpose is to quickly bring the device into operation, maintaining a constant temperature of the case (or inside) LibreVNA. Control via USB with communication with temperature sensors inside LibreVNA, with control from the LibreVNA application (activation by the user of the function at will) or with their regular standalone - whichever is more efficient, then apply.
I have two powerful heatsinks on the bottom and top of the LibreVNA device, and the temperature inside the VNA has a difference with the external environment (above it) within 14 degrees. So, as an option, for example, you can take 45 ... 50 degrees by default as a stably maintained temperature inside LibreVNA with the ability to adjust it. The user will need to install either a radiator or a fan on their own (or as an option to offer with this stove) to ensure that the temperature inside the LibreVNA is less than this limited temperature. Well, then only the algorithm of the "stove" to bring the temperature to the limit and maintain it.
I don’t think it’s worth making such a stove already built into the new LibreVNA modification, because many do not need absolute accuracy or they are satisfied with the option of their own revision of LibreVNA cooling. In general, a modular design is always advantageous because it can have the ability and flexibility to demand individual modules at the user's request.
And in the LibreVNA application, you can make an addition (as a disabled option), until the temperature reaches the set in the settings, the calibration procedure either cannot be started or a temperature mismatch warning appears. And the temperature during the calibration must be stored in the calibration file (or in its name), in order to then control the correspondence of the measurement temperature with this calibration file (within some, probably, ± some degrees).
Best regards, Yury

**Comment by jankae** [comment]
Hi Yury,
thank you very much for the idea. This particular solution hasn't crossed my mind so far. But I doubt I will find the time to develop this rather special addition to the LibreVNA. I am thinking about adding heating resistors to temperature sensitive devices (no need to heat the whole case, just keep critical stuff at a constant temperature). But of course that would only happen in a new version and wouldn't help any current users.
I do encourage you (and other developers) to have a go at your "oven" idea. I think it could be a great addition and am looking forward to pictures and measurement results from such a solution.

**Reply by YuryW** [reply]
Hi Jankae, 
the fact is that in working mode (no heating and cooling, and the temperature is stable within a few degrees) the temperature of the air and all passive elements inside the LibreVNA case will be the same and equal, probably, to the temperature of the case. And in active elements, at their hottest internal points, the temperature will be greater and vary depending on the individual thermal resistances between the internals and the external surface of their bodies.
But the main thing is that the temperature of the chips will be different, but constant, as well as the temperature of their case and with a constant difference with the temperature of the case of the LibreVNA device. So that in this case it doesn't matter which part is stabilized, but it will heat up faster if it is the most heat-inertial parts that are heated, that is, the outer radiators and the LibreVNA body itself, since until their temperature is stabilized, the temperature of all the internals of the LibreVNA will change. And the heater should be several times more powerful than all internal chips, in order to several times accelerate the output of LibreVNA to a stable temperature, which is also very important. That is needed powerful BP.
Therefore the most effective way — to have a separate "stove" with its own powerful BP. Let's see how other users of LibreVNA get inspired. ,
Bad, that is not visible from the outside, are there new posts in this discussion section, probably that's why no one has replied more yet.

Best regards, Yury

**Comment by ghost** [comment]
Hi,

I am a complete beginner to the LibreVNA. I can not find where can I download the software. in https://github.com/jankae/LibreVNA it is written that  after downloading and extracting files I should start LibreVNA-GUI.exe But in v1.4.1 release (latest for now) there is no such file. Can anyone tell, how to download and run the software? Thank you in advance

**Reply by YuryW** [reply]
> how to download and run the software?

Hi Vilius,
The 1.4.0 application itself (you indicated a very important device upgrade) should be viewed below. 
https://github.com/jankae/LibreVNA/releases/tag/v1.4.0
But I would recommend using the instructions for switching to version 1.5 (for now, a test one)
https://github.com/jankae/LibreVNA/releases/tag/v1.5.0-alpha.1
But use the latest app and device firmware updates:
https://github.com/jankae/LibreVNA/actions/runs/4257244823 (at the very bottom of the page)
Best regards.

**Reply by ghost** [reply]
I found it, Thank you very much.
